frontends/p4/sideEffects.h

```diff
@@ -26,15 +26,23 @@ limitations under the License.
 
 namespace P4 {
 
-// Determines whether an expression may have side-effects
-// (but not any sub-expression)
+/** @brief Determines whether an expression may have method or constructor
+ * invocations.
+ *
+ * The TypeMap and ReferenceMap arguments may be null, in which case every
+ * method call expression is counted.  With type information, invocations
```

Comment:
"is assumed to have side-effects". Otherwise said, the analysis may be more conservative.